certbot/docs/contributing.rst

```diff
@@ -134,24 +134,20 @@ You can test your code in several ways:
 Running automated unit tests
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-When you are working in a file ``foo.py``, there should also be a file ``foo_test.py``
-either in the same directory as ``foo.py`` or in the ``tests`` subdirectory
-(if there isn't, make one). While you are working on your code and tests, run
-``python foo_test.py`` to run the relevant tests.
+To run all unittests:
 
-For debugging, we recommend putting
-``import ipdb; ipdb.set_trace()`` statements inside the source code.
+    tox
+
+If you're working on a specific test and would like to run just that one:
+
+    pytest acme/acme/_internal/tests/messages_test.py # Use the test file you're working on
 
-Once you are done with your code changes, and the tests in ``foo_test.py``
-pass, run all of the unit tests for Certbot and check for coverage with ``tox
--e cover``. You should then check for code style with ``tox run -e lint`` (all
-files) or ``pylint --rcfile=.pylintrc path/to/file.py`` (single file at a
-time).
+To run a specific test case within a file:
 
-Once all of the above is successful, you may run the full test suite using
-``tox --skip-missing-interpreters``. We recommend running the commands above
-first, because running all tests like this is very slow, and the large amount
-of output can make it hard to find specific failures when they happen.
+    pytest acme/acme/_internal/tests/messages_test.py::ChallengeBodyTest::test_to_partial_json
```

Comment:
```suggestion
.. code-block:: shell

    pytest acme/acme/_internal/tests/messages_test.py -k test_to_partial_json
```

The other syntax is fine but imo this one is easier, it's what I personally use. Feel free to ignore.